lib/money/money.rb

```diff
@@ -207,11 +207,20 @@ def to_d
   end
 
   def to_s(style = nil)
-    case style
+    units = case style
     when :legacy_dollars
-      sprintf("%.2f", value)
+      2
     when :amount, nil
-      sprintf("%.#{currency.minor_units}f", value)
+      currency.minor_units
+    else
+      return nil
+    end
+
+    rounded = round(units)
```

Comment:
the `value` is already rounded. How about
```ruby
def to_s(style = nil)
  units = (style == :legacy_dollars ? 2 : currency.minor_units)
  if units == 0
    sprintf("%d", value)
  else
    sprintf("%d.%0#{units}d", value.truncate, value.frac * (10 ** units))
  end
end
```

Reply:
The value isn't always already rounded. Example is `Money.new("1.999", "USD)` (or possibly `Money.new("1.99", "USD) / 2`. These amounts have 3 decimals, and on master they are rendered as `"2.00"` and `"1.00"`, respectively. We need to round for that.

`units = (style == :legacy_dollars ? 2 : currency.minor_units)` is slightly breaking. The old version returns `nil` if `style` is an unexpected value like `:foo`. That's why my code looks like it does. I'm happy to change to what you propose if we're fine with the breaking change.

Reply:
sorry i meant the value is rounded when initialized here https://github.com/Shopify/money/blob/master/lib/money/money.rb#L87. You're right though that this won't work for a 3 digit currency we're trying to display using "legacy_dollars" notation. 

I belive you can remove the `to_d` here since these would already all be decimals

Reply:
Right, I can change `rounded = round(units)` to `rounded_value = @value.round(units)`, and skip the `to_d` parts.